### PR TITLE
HW5: create server

### PR DESCRIPTION
**Pull request by YuliaBarbashova**

### CODE REVIEW

File: HW5/controllers/hobbiesController.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      };
+      res.writeHead(200, {
+        "Cache-Control": "private, max-age=3600",
+        "Content-Type": "application/json",
+      });
+      res.write(JSON.stringify(responseObject));
+      res.end();
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (AlexeyBarkow) [inline HW5/controllers/hobbiesController.js]: I'd rather return 500 status code in this case

File: HW5/controllers/hobbiesController.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        links: {
+          self: `${PATH}/${userHobbies.userId}/hobbies`,
+          user: `${PATH}/${userHobbies.userId}`,
+        },
+      };
+      const responseObject = {
+        data: formatedData,
+        error: null,
+      };
+      res.writeHead(200, {
```
Comment (AlexeyBarkow) [inline HW5/controllers/hobbiesController.js]: I'd move such repeatable code pieces in a separate utility helper

File: HW5/server.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+} = require("./controllers/usersController");
+const { PATH } = require("./constants");
+
+const server = http.createServer((req, res) => {
+  if (req.method === "GET" && req.url === PATH) {
+    getUsers(req, res);
+  } else if (
+    req.method === "GET" &&
+    req.url.indexOf(PATH) === 0 &&
+    req.url.indexOf("/hobbies") > 0
```
Comment (AlexeyBarkow) [inline HW5/server.js]: such logic allows urls with extra symbols between PATH and hobbies, and after /hobbies:)
```
http://localhost:8000/api/users/b2502e61-5557-408a-b39d-e4abe7c000d7/test/hobbies-blah/test
```

File: HW5/controllers/hobbiesController.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      } else {
+        newHobbies = Array.from(new Set([...userHobbies.hobbies, ...hobbies]));
+        await HobbiesModel.updateUserHobbies(user.id, newHobbies);
+        const formatedData = formatUserData(user);
+        const responseObject = {
+          data: formatedData,
+          error: null,
+        };
+        res.statusCode = 200;
+        res.setHeader("Content-Type", "aplication/json");
```
Comment (AlexeyBarkow) [inline HW5/controllers/hobbiesController.js]: typo in a word application

File: HW5/models/userModel.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+const { v4: uuidv4 } = require("uuid");
+const path = require("path");
+
+const users = require("../data/users.json");
```
Comment (AlexeyBarkow) [inline HW5/models/userModel.js]: Unfortunately, when you update a file, it doesn't update imported objects
Thus, when you add a new user, it isn't returned from getAllUsers method without restarting the server

File: HW5/utils.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+const fs = require("fs");
+
+const { PATH } = require("./constants");
+
+const writeDataInFile = (fileName, data) => {
+  fs.writeFileSync(fileName, JSON.stringify(data), "utf8", (err) =>
```
Comment (AlexeyBarkow) [inline HW5/utils.js]: FYI: you can structurize your json data in file using JSON.stringify 3rd argument: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#space

**Review by AlexeyBarkow — CHANGES_REQUESTED** [review]

File: HW5/controllers/hobbiesController.js (hunk 1 above)
Comment (YuliaBarbashova) [reply HW5/controllers/hobbiesController.js]: Updated

File: HW5/controllers/hobbiesController.js (hunk 2 above)
Comment (YuliaBarbashova) [reply HW5/controllers/hobbiesController.js]: Added

File: HW5/server.js (hunk 3 above)
Comment (YuliaBarbashova) [reply HW5/server.js]: Updated

File: HW5/controllers/hobbiesController.js (hunk 4 above)
Comment (YuliaBarbashova) [reply HW5/controllers/hobbiesController.js]: Fixed

File: HW5/models/userModel.js (hunk 5 above)
Comment (YuliaBarbashova) [reply HW5/models/userModel.js]: Thank you, updated

File: HW5/utils.js (hunk 6 above)
Comment (YuliaBarbashova) [reply HW5/utils.js]: Updated

**Review by AlexeyBarkow — APPROVED** [review]

**Pull request merged**